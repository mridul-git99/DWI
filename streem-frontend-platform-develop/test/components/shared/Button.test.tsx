import { Button } from '#components';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

describe('Button component', () => {
  it('renders button with default props', () => {
    const { getByText } = render(<Button>Default Button</Button>);
    const buttonElement = getByText('Default Button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement.tagName).toBe('BUTTON');
    expect(buttonElement).toHaveStyle('background-color: #1d84ff');
    expect(buttonElement).toHaveStyle('color: #ffffff');
    expect(buttonElement).toMatchSnapshot();
    // Add more assertions for default props if needed
  });

  it('renders button with specified color and variant', () => {
    const { getByText } = render(
      <Button color="green" variant="secondary">
        Secondary Green Button
      </Button>,
    );
    const buttonElement = getByText('Secondary Green Button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveStyle('background-color: #ffffff');
    expect(buttonElement).toHaveStyle('border-color: #5aa700');
    expect(buttonElement).toHaveStyle('color: #5aa700');
    expect(buttonElement).toMatchSnapshot();
    // Add more assertions for the specified color and variant
  });

  it('handles click event', () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button onClick={handleClick}>Click Me</Button>);
    const buttonElement = getByText('Click Me');
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(buttonElement).toMatchSnapshot();
    // Add more assertions for the click event if needed
  });

  it('renders text-only button with transparent background', () => {
    const { getByText } = render(
      <Button variant="textOnly" color="red">
        Text-only Button
      </Button>,
    );
    const buttonElement = getByText('Text-only Button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveStyle('background-color: transparent');
    expect(buttonElement).toHaveStyle('color: #ff6b6b');
    expect(buttonElement).toMatchSnapshot();
    // Add more assertions for the text-only variant if needed
  });

  it('renders disabled secondary button', () => {
    const { getByText } = render(
      <Button disabled variant="secondary">
        Disabled Button
      </Button>,
    );
    const buttonElement = getByText('Disabled Button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toBeDisabled();
    expect(buttonElement).toHaveStyle('background-color: #fff');
    expect(buttonElement).toHaveStyle('color: #bbbbbb');
    expect(buttonElement).toMatchSnapshot();
    // Add more assertions for the disabled secondary button if needed
  });
});
