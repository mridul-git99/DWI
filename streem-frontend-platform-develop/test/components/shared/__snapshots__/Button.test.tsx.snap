// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component handles click event 1`] = `
.c0 {
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 24px;
  outline: none;
  padding-block: 12px;
  padding-inline: 24px;
  background-color: #1d84ff;
  color: #ffffff;
}

.c0:last-of-type {
  margin-right: 0;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 > .icon {
  color: #ffffff;
}

.c0:hover {
  background-color: #005dcc;
}

.c0:active {
  background-color: #00387a;
}

<button
  class="c0"
  type="button"
>
  Click Me
</button>
`;

exports[`Button component renders button with default props 1`] = `
.c0 {
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 24px;
  outline: none;
  padding-block: 12px;
  padding-inline: 24px;
  background-color: #1d84ff;
  color: #ffffff;
}

.c0:last-of-type {
  margin-right: 0;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 > .icon {
  color: #ffffff;
}

.c0:hover {
  background-color: #005dcc;
}

.c0:active {
  background-color: #00387a;
}

<button
  class="c0"
  type="button"
>
  Default Button
</button>
`;

exports[`Button component renders button with specified color and variant 1`] = `
.c0 {
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 24px;
  outline: none;
  padding-block: 12px;
  padding-inline: 24px;
  background-color: #ffffff;
  border-color: #5aa700;
  color: #5aa700;
}

.c0:last-of-type {
  margin-right: 0;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 > .icon {
  color: inherit;
}

.c0:hover {
  background-color: #5aa700;
  color: #ffffff;
}

.c0:hover > .icon {
  color: inherit;
}

.c0:active {
  background-color: #427a00;
}

<button
  class="c0"
  color="green"
  type="button"
>
  Secondary Green Button
</button>
`;

exports[`Button component renders disabled secondary button 1`] = `
.c0 {
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 24px;
  outline: none;
  padding-block: 12px;
  padding-inline: 24px;
  background-color: #ffffff;
  border-color: #1d84ff;
  color: #1d84ff;
  border-color: #bbbbbb;
  color: #bbbbbb;
  background-color: #fff;
}

.c0:last-of-type {
  margin-right: 0;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 > .icon {
  color: inherit;
}

.c0:hover {
  background-color: #005dcc;
  color: #ffffff;
}

.c0:hover > .icon {
  color: inherit;
}

.c0:active {
  background-color: #00387a;
}

.c0:hover {
  background-color: #fff;
  color: #bbbbbb;
}

<button
  class="c0"
  disabled=""
  type="button"
>
  Disabled Button
</button>
`;

exports[`Button component renders text-only button with transparent background 1`] = `
.c0 {
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 24px;
  outline: none;
  padding-block: 12px;
  padding-inline: 24px;
  background-color: transparent;
  color: #ff6b6b;
  padding: 4px 8px;
}

.c0:last-of-type {
  margin-right: 0;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:hover,
.c0:active {
  background-color: #fafafa;
}

<button
  class="c0"
  color="red"
  type="button"
>
  Text-only Button
</button>
`;
